src/java.base/unix/classes/jdk/internal/loader/FileURLMapper.java

```diff
@@ -40,41 +42,44 @@
  * @author      Michael McMahon
  */
 
-public class FileURLMapper {
+final class FileURLMapper {
 
-    URL url;
-    String path;
+    private final URL url;
+    private String path;
 
-    public FileURLMapper (URL url) {
+    FileURLMapper(URL url) {
         this.url = url;
     }
 
     /**
      * @return the platform specific path corresponding to the URL
      *  so long as the URL does not contain a hostname in the authority field.
      */
-
-    public String getPath () {
+    String getPath() throws IOException {
         if (path != null) {
             return path;
         }
         String host = url.getHost();
         if (host == null || host.isEmpty() || "localhost".equalsIgnoreCase(host)) {
             path = url.getFile();
-            path = ParseUtil.decode(path);
+            try {
+                path = ParseUtil.decode(path);
+            } catch (IllegalArgumentException iae) {
+                throw new IOException(iae);
+            }
         }
         return path;
     }
 
     /**
      * Checks whether the file identified by the URL exists.
      */
-    public boolean exists () {
-        String s = getPath ();
+    boolean exists() throws IOException {
+        String s = getPath();
         if (s == null) {
             return false;
         } else {
-            File f = new File (s);
+            File f = new File(s);
```

Comment:
Lots of unrelated cleanup in this file, but okay.

Reply:
Right, I was considering of not doing it, but the number of such lines didn't seem too many, so decided to include it. If you and others lean towards leaving the formatting clean up out of this, then I will update the PR accordingly.